tests/load/locustfile.py

```diff
@@ -4,42 +4,91 @@
-        json_path = os.path.join(project_root, "tests/load/assets/test_data.json")
-
-        with open(json_path, "r") as f:
-            return json.load(f)
-
     def on_start(self):
+        # Check if the database has been seeded before starting the test
+        self.__class__.seeded = self.environment.parsed_options.seeded
+
         with self.seed_lock:
             if not self.__class__.seeded:
-                self.__class__.seed_data = self.load_seed_data()
-                # TODO: Adjust to loop through the seed data; API endpoint has a limit of 100 clusters
-                self.client.post("/api/seed", json=self.__class__.seed_data)
-                self.__class__.seeded = True
+                locust_file = pathlib.Path(__file__).resolve()
+                project_root = locust_file.parents[2]
+                seed_data_path = os.path.join(project_root, "tests/load/assets/seed_data.json")
+                with open(seed_data_path, "rb") as input_file:
+                    clusters_iter = ijson.items(input_file, "clusters.item", use_float=True)
+                    chunk = []
+                    chunk_size = 100
+                    idx = 0
+                    for cluster in clusters_iter:
+                        chunk.append(cluster)
+                        if len(chunk) == chunk_size:
+                            idx += chunk_size
+                            seed_data = json.dumps({"clusters": chunk}, indent=2)
+                            chunk = []
 
-    def pick_random_record(self):
-        # Pick a random record from the seed data
-        rando = random.choice(self.__class__.seed_data["clusters"])
-        return rando
+                    if chunk:
+                        seed_data = json.dumps({"clusters": chunk}, indent=2)
+                        self.client.post("/api/seed", seed_data)
+                self.__class__.seeded = True
 
     @locust.task
     def link(self):
-        random_data = self.pick_random_record()
-        data = {
-            "record": random_data["records"][0],
-            "external_person_id": random_data["external_person_id"],
-        }
-        self.client.post("/api/link", json=data)
-
-        print(f"Linking...{random_data['external_person_id']}")
+        """
+        Loops through the original data, randomly selects records, and links them.
+        """
+        records_to_link = self.environment.parsed_options.records_to_link
+
+        # Pick a random record from the seed data
+        locust_file = pathlib.Path(__file__).resolve()
+        project_root = locust_file.parents[2]
+        original_data_path = os.path.join(project_root, "tests/load/assets/original_data.json")
+
+        with open(original_data_path, "rb") as input_file:
+            clusters_iter = ijson.items(input_file, "clusters.item", use_float=True)
+            counter = 1
+            for cluster in clusters_iter:
+                if random.random() < 0.5:  # randomly decide whether to link or not
```

Comment:
With this implementation, we'll send about 50% of the data to be linked.  What do you think about making this a variable?